dartagnan/src/main/java/com/dat3m/dartagnan/encoding/EncodingContext.java

```diff
@@ -249,6 +261,9 @@ public BooleanFormula equal(Formula left, Formula right) {
         if (left instanceof BooleanFormula l && right instanceof BooleanFormula r) {
             return booleanFormulaManager.equivalence(l, r);
         }
+        if (left instanceof TupleFormula l && right instanceof TupleFormula r) {
+            return new TupleFormulaManager(this).equal(l, r);
```

Comment:
with do we need to create a new instance and not simply use `tupleFormulaManager`?